### ISSUE TITLE
Upgrade

### ISSUE THREAD

**Issue by DJPille1** [Percedal/FS22_UpgradableFactories#48]
Would it be possible to make it compatible with fedaction's productions?

**Comment by berntseb** [comment]
It is, I am using it with fedaction productions since ever.
The only challenge: Once FedAction updates his production, adds new inputs or output, you need to manually add those fillTypes to the upgradableFactories.xml in your savegame. At least with value "0". As long as the input is not present, production won't work. If the new output filltype is not present, the game will crash at the first production cycle.

**Comment by Shad0wlife** [comment]
Huh this would actually be a sensible thing to add in a fix; checking for new filltypes somehow.

**Comment by berntseb** [comment]
I am not too familiar with the Giants Engine. So this is actually where I struggle: Do the same in code, as I do manually - Compare the productions (manually in placeables.xml) and its input and output filltypes with those in the upgradeableFactories.xml

**Comment by berntseb** [comment]
I guess one of the main issue of the mod has always been that it stores the upgrades in a separate File instead of manipulating the productions in the placeables.xml

**Comment by Shad0wlife** [comment]
Yes, but that is "needed" because the placeable xml System is given by the game and I don't think it's "safe" to change that.

**Comment by Shad0wlife** [comment]
Huh, it actually should save everything that has a storage in a building to the xml just fine, it just iterates through all the storages. It just won't load a new storage fill level, since that is missing, but the next time it's saved it should turn up. And the capacity is also generated on load and not taken from the file (that only holds the fill level anyways I think).

So I have no idea why new storage types wouldn't show up after an update...
